frontend/pages/SoftwarePage/SoftwareAddPage/SoftwareAppStoreVpp/SoftwareVppForm/SoftwareVppForm.tsx

```diff
@@ -0,0 +1,318 @@
+  labels,
+  vppApps,
+  softwareVppForEdit,
+  onSubmit,
+  isUploading = false,
+  onCancel,
+}: ISoftwareVppFormProps<T>) => {
+  const [formData, setFormData] = useState<ISoftwareVppFormData>(
+    softwareVppForEdit
+      ? {
+          selfService: softwareVppForEdit.self_service || false,
+          targetType: "All hosts",
+          customTarget: "labelsIncludeAny",
+          labelTargets: {},
+        }
+      : {
+          selectedApp: null,
+          selfService: false,
+          targetType: "All hosts",
+          customTarget: "labelsIncludeAny",
+          labelTargets: {},
+        }
+  );
+
+  const [formValidation, setFormValidation] = useState<IFormValidation>({
+    isValid: !!softwareVppForEdit, // Disables submit before VPP to add is selected
+  });
+
+  const onFormSubmit = (evt: React.FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+    onSubmit(formData);
+  };
+
+  const onSelectApp = (app: IVppApp) => {
+    if ("selectedApp" in formData) {
+      const newFormData = {
+        ...formData,
+        selectedApp: app,
+        selfService:
+          app.platform === "ios" || app.platform === "ipados"
+            ? false
+            : formData.selfService,
+      };
+      setFormData(newFormData);
+      setFormValidation(generateFormValidation(newFormData));
+    }
+  };
+
+  const onSelectTargetType = (value: string) => {
+    const newData = { ...formData, targetType: value };
+    setFormData(newData);
+    setFormValidation(generateFormValidation(newData));
+  };
+
+  const onSelectCustomTargetOption = (value: string) => {
+    const newData = { ...formData, customTarget: value };
+    setFormData(newData);
+  };
+
+  const onSelectLabel = ({ name, value }: { name: string; value: boolean }) => {
+    const newData = {
+      ...formData,
+      labelTargets: { ...formData.labelTargets, [name]: value },
+    };
+    setFormData(newData);
+    setFormValidation(generateFormValidation(newData));
+  };
+
+  const isSubmitDisabled = !formValidation.isValid;
+
+  const renderSelfServiceContent = (platform: string) => {
+    if (platform !== "ios" && platform !== "ipados") {
+      return (
+        <Checkbox
+          value={formData.selfService}
+          onChange={(newVal: boolean) =>
+            setFormData({ ...formData, selfService: newVal })
+          }
+          className={`${baseClass}__self-service-checkbox`}
+          tooltipContent={
+            <>
+              End users can install from <b>Fleet Desktop</b> {">"}{" "}
+              <b>Self-service</b>.
+            </>
+          }
+        >
+          Self-service
+        </Checkbox>
+      );
+    }
+    return null;
+  };
+
+  const renderContent = () => {
+    if (softwareVppForEdit) {
+      return (
+        <div className={`${baseClass}__form-fields`}>
+          <FileUploader
+            canEdit={false}
+            graphicName={"app-store"}
+            message=".pkg, .msi, .exe, .deb, or .rpm"
```

Comment:
Seems like we should remove this line as it's not applicable?

Reply:
Changed to `FileDetails` component, thanks for catching!